internal/resource-request-operator/broadcaster.go

```diff
@@ -6,46 +6,72 @@ import (
 	"sync"
 	"time"
 
-	crdclient "github.com/liqotech/liqo/pkg/crdClient"
-
-	configv1alpha1 "github.com/liqotech/liqo/apis/config/v1alpha1"
-	"github.com/liqotech/liqo/pkg/utils"
-
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog"
+	resourcehelper "k8s.io/kubectl/pkg/util/resource"
+
+	configv1alpha1 "github.com/liqotech/liqo/apis/config/v1alpha1"
+	crdclient "github.com/liqotech/liqo/pkg/crdClient"
+	"github.com/liqotech/liqo/pkg/utils"
+	"github.com/liqotech/liqo/pkg/virtualKubelet/forge"
 )
 
 // Broadcaster is an object which is used to get resources of the cluster.
 type Broadcaster struct {
-	allocatable   corev1.ResourceList
-	clusterConfig configv1alpha1.ClusterConfig
-	offerMutex    sync.RWMutex
-	configMutex   sync.RWMutex
-	informer      cache.SharedInformer
+	allocatable    corev1.ResourceList
+	resourcePodMap map[string]corev1.ResourceList
+	clusterConfig  configv1alpha1.ClusterConfig
+	nodeMutex      sync.RWMutex
+	podMutex       sync.RWMutex
+	configMutex    sync.RWMutex
+	nodeInformer   cache.SharedIndexInformer
+	podInformer    cache.SharedIndexInformer
 }
 
 // SetupBroadcaster create the informer e run it to signal node changes updating Offers.
-func (b *Broadcaster) SetupBroadcaster(clientset *kubernetes.Clientset, resyncPeriod time.Duration) error {
+func (b *Broadcaster) SetupBroadcaster(clientset kubernetes.Interface, resyncPeriod time.Duration) error {
 	b.allocatable = corev1.ResourceList{}
+	b.resourcePodMap = map[string]corev1.ResourceList{}
 	factory := informers.NewSharedInformerFactory(clientset, resyncPeriod)
-	b.informer = factory.Core().V1().Nodes().Informer()
-	b.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    b.onAdd,
-		UpdateFunc: b.onUpdate,
-		DeleteFunc: b.onDelete,
+	b.nodeInformer = factory.Core().V1().Nodes().Informer()
+	b.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    b.onNodeAdd,
+		UpdateFunc: b.onNodeUpdate,
+		DeleteFunc: b.onNodeDelete,
+	})
+
+	factoryWithOptions := informers.NewSharedInformerFactoryWithOptions(clientset, resyncPeriod, informers.WithTweakListOptions(setLabelSelectors))
+	b.podInformer = factoryWithOptions.Core().V1().Pods().Informer()
+	b.podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    b.onPodAdd,
+		DeleteFunc: b.onPodDelete,
 	})
 
 	return nil
 }
 
-// StartBroadcaster starts a new sharedInformer to watch nodes resources.
+// StartBroadcaster starts two shared Informers, one for nodes and one for pods launching two separated goroutines.
 func (b *Broadcaster) StartBroadcaster(ctx context.Context, group *sync.WaitGroup) {
+	group.Add(2)
```

Comment:
I suppose it is more idiomatic to increment this in the actual functions, just before deferring the Done